<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1286</width>
    <height>641</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <property name="styleSheet">
   <string notr="true">/*background: qradialgradient(spread:pad, cx:0.5, cy:0.5, radius:0.5, fx:0.5, fy:0.5, stop:0 rgba(255, 235, 235, 206), stop:0.35 rgba(255, 188, 188, 80), stop:0.4 rgba(255, 162, 162, 80), stop:0.425 rgba(255, 132, 132, 156), stop:0.44 rgba(252, 128, 128, 80), stop:1 rgba(255, 255, 255, 0))*/

</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="23" column="2">
     <widget class="QLabel" name="label_2">
      <property name="text">
       <string>Comparisons</string>
      </property>
     </widget>
    </item>
    <item row="16" column="1">
     <widget class="QComboBox" name="InitiateOrder">
      <item>
       <property name="text">
        <string>Random</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Almost Sorted</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Almost Sorted in Reverse Order</string>
       </property>
      </item>
     </widget>
    </item>
    <item row="10" column="1">
     <widget class="QSlider" name="delay">
      <property name="minimumSize">
       <size>
        <width>30</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>1000</number>
      </property>
      <property name="singleStep">
       <number>10</number>
      </property>
      <property name="pageStep">
       <number>10</number>
      </property>
      <property name="value">
       <number>500</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="tickPosition">
       <enum>QSlider::TicksBelow</enum>
      </property>
      <property name="tickInterval">
       <number>100</number>
      </property>
     </widget>
    </item>
    <item row="3" column="1">
     <widget class="QLabel" name="BestCase">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="21" column="1" rowspan="2">
     <widget class="QGroupBox" name="VisualizationControl">
      <property name="title">
       <string>Visualization control</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QPushButton" name="StopButton">
         <property name="text">
          <string>Play/Pause</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="ClearButton">
         <property name="text">
          <string>Clear</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="1">
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="4" column="1">
     <widget class="QLabel" name="Average">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="15" column="1">
     <widget class="QLabel" name="label_7">
      <property name="text">
       <string>Initiate order</string>
      </property>
     </widget>
    </item>
    <item row="9" column="1">
     <widget class="QSpinBox" name="numberOfItems">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true">background: rgb(255, 255, 255)</string>
      </property>
      <property name="minimum">
       <number>2</number>
      </property>
      <property name="maximum">
       <number>10000000</number>
      </property>
      <property name="value">
       <number>100</number>
      </property>
     </widget>
    </item>
    <item row="13" column="1">
     <widget class="QLabel" name="label_6">
      <property name="text">
       <string>Sorting order</string>
      </property>
     </widget>
    </item>
    <item row="23" column="5">
     <widget class="QLineEdit" name="writes"/>
    </item>
    <item row="7" column="1">
     <widget class="QLabel" name="label_3">
      <property name="maximumSize">
       <size>
        <width>130</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(255, 255, 255);</string>
      </property>
      <property name="text">
       <string>Amount of items</string>
      </property>
     </widget>
    </item>
    <item row="6" column="1">
     <widget class="QPushButton" name="ComplexityButton">
      <property name="text">
       <string>Computational complexity</string>
      </property>
     </widget>
    </item>
    <item row="19" column="1">
     <widget class="QComboBox" name="SortingNameComboBox">
      <property name="minimumSize">
       <size>
        <width>300</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>16777215</height>
       </size>
      </property>
      <item>
       <property name="text">
        <string>BubbleSort</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>InsertionSort</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>SelectionSort</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>CycleSort</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>ShakerSort</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>CombSort</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>GnomeSort</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>OddEvenSort</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>QuickSortPivotFirst</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>QuickSortPivotLast</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>QuickSortPivotMiddle</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>QuickSortPivotRandom</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>MergeSort</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>MergeSortInPlace</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>HeapSort</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>TimSort</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>IntroSort</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>ShellSort</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>PigeonholeSort</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>BucketSort</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>CountingSort</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>RadixSort</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>FlashSort</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>PancakeSort</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>BogoSort</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>StoogeSort</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>SlowSort</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>TreeSort</string>
       </property>
      </item>
     </widget>
    </item>
    <item row="23" column="7">
     <widget class="QLineEdit" name="reads">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="20" column="1">
     <widget class="QGroupBox" name="groupBox">
      <property name="minimumSize">
       <size>
        <width>300</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true">background: rgb(34,193,195);
background: linear-gradient(0deg, rgba(34,193,195,1) 0%, rgba(149,45,253,0.38728995015975143) 100%);</string>
      </property>
      <property name="title">
       <string/>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QPushButton" name="InitiateButton">
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>Initiate</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="SortButton">
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>Sort</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="23" column="4">
     <widget class="QLabel" name="label_4">
      <property name="text">
       <string>Writes</string>
      </property>
     </widget>
    </item>
    <item row="0" column="2" rowspan="23" colspan="7">
     <widget class="QGraphicsView" name="graphicsView">
      <property name="styleSheet">
       <string notr="true">background: rgb(255, 255, 255)</string>
      </property>
     </widget>
    </item>
    <item row="23" column="1">
     <widget class="QLabel" name="SortingTime">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="23" column="6">
     <widget class="QLabel" name="label_5">
      <property name="text">
       <string>Reads</string>
      </property>
     </widget>
    </item>
    <item row="17" column="1">
     <widget class="QLabel" name="label_8">
      <property name="text">
       <string>Sorting type</string>
      </property>
     </widget>
    </item>
    <item row="14" column="1">
     <widget class="QComboBox" name="SortingOrder">
      <item>
       <property name="text">
        <string>Increasing</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Decreasing</string>
       </property>
      </item>
     </widget>
    </item>
    <item row="23" column="3">
     <widget class="QLineEdit" name="comparisons"/>
    </item>
    <item row="11" column="1">
     <widget class="QLabel" name="label">
      <property name="maximumSize">
       <size>
        <width>110</width>
        <height>30</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(255, 255, 255);</string>
      </property>
      <property name="text">
       <string>Visualization Delay</string>
      </property>
     </widget>
    </item>
    <item row="5" column="1">
     <widget class="QLabel" name="WorstCase">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1286</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>delay</sender>
   <signal>valueChanged(int)</signal>
   <receiver>label</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>53</x>
     <y>420</y>
    </hint>
    <hint type="destinationlabel">
     <x>61</x>
     <y>448</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
